### PR TITLE
E2E Scenarios 6.2.1 and 6.2.3

### PR DESCRIPTION
**Pull request by sergiishamrai-okta**

### CODE REVIEW

File: direct-auth-samples/pom.xml [hunk 1]
```diff
@@ -167,6 +167,24 @@
             <version>5.3.5</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Comment (arvindkrishnakumar-okta) [inline direct-auth-samples/pom.xml:170]: Add placeholder property variable for version `4.0.0`

File: direct-auth-samples/src/test/java/env/Hooks.java [hunk 2]
```diff
@@ -16,17 +16,35 @@
...
+import org.slf4j.LoggerFactory;
 import pages.Page;
 
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
 public class Hooks {
+
+	private final Logger logger = LoggerFactory.getLogger(Hooks.class);
```
Comment (arvindkrishnakumar-okta) [inline direct-auth-samples/src/test/java/env/Hooks.java:43]:
```suggestion
	private static final Logger logger = LoggerFactory.getLogger(Hooks.class);
```

File: direct-auth-samples/src/test/java/info/seleniumcucumber/userStepDefinitions/MFA.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.Assert;
+import org.openqa.selenium.WebDriver;
+import pages.*;
+
+public class MFA extends CucumberRoot {
+
+    private static final int RETRY_COUNT = 5;
```
Comment (arvindkrishnakumar-okta) [inline direct-auth-samples/src/test/java/info/seleniumcucumber/userStepDefinitions/MFA.java:31]: Add it to configuration?

File: direct-auth-samples/src/test/java/pages/RootPage.java [hunk 4]
```diff
@@ -60,6 +60,11 @@ public void navigateToTheRootPage() {
         driver.get("http://localhost:8080"); // TODO pass as env variable.
     }
 
+    public void navigateToThePasswordResetPage() {
+        driver.manage().window().maximize();
+        driver.get("http://localhost:8080/forgot-password"); // TODO pass as env variable.
```
Comment (arvindkrishnakumar-okta) [inline direct-auth-samples/src/test/java/pages/RootPage.java:65]: Please create a followup/backlog JIRA to address this.

**Review by arvindkrishnakumar-okta — APPROVED** [review]
minor changes requested.

File: direct-auth-samples/src/test/java/info/seleniumcucumber/userStepDefinitions/MFA.java (hunk 3 above)
Comment (sergiishamrai-okta) [reply direct-auth-samples/src/test/java/info/seleniumcucumber/userStepDefinitions/MFA.java:31]: Int ref: [OKTA-401435](https://oktainc.atlassian.net/browse/OKTA-401435)

File: direct-auth-samples/src/test/java/pages/RootPage.java (hunk 4 above)
Comment (sergiishamrai-okta) [reply direct-auth-samples/src/test/java/pages/RootPage.java]: Int ref:  [OKTA-401435](https://oktainc.atlassian.net/browse/OKTA-401435)

**Pull request merged**